src/AbstractOperations/computations.jl

```diff
@@ -37,17 +39,20 @@ Perform a `computation`. The result is stored in `computation.result`.
 function compute!(computation::Computation)
     arch = architecture(computation.result)
     result_data = data(computation.result)
-    @launch(device(arch), config=launch_config(computation.grid, :xyz),
-            _compute!(result_data, computation.grid, computation.operation))
+
+    workgroup, worksize= work_layout(computation.grid, :xyz)
+    kernel! = _compute!(device(arch), workgroup, worksize)
+
+    event = kernel!(result_data, computation.grid, computation.operation)
+    wait(event)
```

Comment:
This blocks progress, and will cause Oceanigans to be slower than before. If you don't want to deal with the full dependency system you can do:

```
kernel! = _compute!(device(arch), workgroup, worksize)

event = kernel!(result_data, computation.grid, computation.operation, 
                         dependencies=Event(device(arch)))
wait(device(arch), event)
```

Which will synchronize everything through the global stream.